tests/unit_tests/models/test_model_bridge_lora.py

```diff
@@ -375,6 +392,253 @@ def megatron_to_hf(self, weight, module):
+
+    qkv_slices = {
+        "q_proj": torch.full((2, 2), 1.0),
+        "k_proj": torch.full((2, 2), 2.0),
+        "v_proj": torch.full((2, 2), 3.0),
+    }
+    monkeypatch.setattr(bridge, "_split_qkv_linear_out_weight", lambda *_: qkv_slices)
+
+    weights = list(
+        bridge.stream_adapter_weights_megatron_to_hf(
+            [SimpleNamespace(config=SimpleNamespace())],
+            cpu=False,
+            show_progress=False,
+        )
+    )
+
+    assert len(weights) == 6
+    names = [w.param_name for w in weights]
+    assert names[0].endswith("q_proj.lora_A.weight")
+    assert names[1].endswith("q_proj.lora_B.weight")
+    assert names[2].endswith("k_proj.lora_A.weight")
+    assert names[3].endswith("k_proj.lora_B.weight")
+    assert names[4].endswith("v_proj.lora_A.weight")
+    assert names[5].endswith("v_proj.lora_B.weight")
+
+    torch.testing.assert_close(weights[0].weight, torch.ones(2, 2))
+    torch.testing.assert_close(weights[1].weight, qkv_slices["q_proj"])
+    torch.testing.assert_close(weights[3].weight, qkv_slices["k_proj"])
+
+
+def test_stream_adapter_weights_megatron_to_hf_fused_fc1(monkeypatch):
+    bridge = DummyBridge()
+
+    adapter_task = AdapterWeightConversionTask(
+        global_base_prefix="decoder.layers.0.mlp.linear_fc1",
+        adapter_key=None,
+        alpha=2,
+        dim=4,
+        linear_in_task=WeightConversionTask(
+            param_name="local_in",
+            global_param_name="decoder.layers.0.mlp.linear_fc1.adapter.linear_in.weight",
+            mapping=Mock(),
+        ),
+        linear_out_task=WeightConversionTask(
+            param_name="local_out",
+            global_param_name="decoder.layers.0.mlp.linear_fc1.adapter.linear_out.weight",
+            mapping=Mock(),
+        ),
+    )
+
+    # Megatron stores fused FC1 as [gate; up] along dim=0. Streaming should split B
+    # and duplicate A for gate_proj and up_proj.
+    linear_out = torch.cat([torch.full((2, 2), 1.0), torch.full((2, 2), 2.0)], dim=0)
+    adapter_weight = AdapterWeight(
+        global_base_prefix="decoder.layers.0.mlp.linear_fc1",
+        adapter_key=None,
+        alpha=2,
+        dim=4,
+        linear_in_weight=MegatronWeightTuple("local_in", torch.ones(2, 2), vp_stage=0),
+        linear_out_weight=MegatronWeightTuple("local_out", linear_out, vp_stage=0),
+    )
+
+    monkeypatch.setattr(
+        bridge,
+        "build_adapter_conversion_tasks",
+        lambda *_: {"decoder.layers.0.mlp.linear_fc1": [adapter_task]},
+    )
+    monkeypatch.setattr(bridge, "materialize_adapter_weights", lambda *_: [adapter_weight])
+    monkeypatch.setattr(
+        bridge,
+        "_get_base_hf_weight_names_for_adapter",
+        lambda *_: [
+            "model.layers.0.mlp.gate_proj.weight",
+            "model.layers.0.mlp.up_proj.weight",
+        ],
+    )
+
+    weights = list(
+        bridge.stream_adapter_weights_megatron_to_hf(
+            [SimpleNamespace(config=SimpleNamespace())],
```

Comment:
To fix the test cases:

```suggestion
            [SimpleNamespace(config=SimpleNamespace(num_moe_experts=0))],
```